app/lib/activitypub/activity/create.rb

```diff
@@ -165,40 +165,77 @@ def in_reply_to_uri
   end
 
   def text_from_content
+    return Formatter.instance.linkify([text_from_name, object_url].join(' ')) if converted_object_type?
+
     if @object['content'].present?
       @object['content']
-    elsif language_map?
+    elsif content_language_map?
       @object['contentMap'].values.first
     end
   end
 
-  def language_from_content
-    return LanguageDetector.instance.detect(text_from_content, @account) unless language_map?
-    @object['contentMap'].keys.first
+  def text_from_name
+    if @object['name'].present?
+      @object['name']
+    elsif name_language_map?
+      @object['nameMap'].values.first
+    end
+  end
```

Comment:
can be `@object['name] || @object['nameMap']&.values&.first` but it feels like we need a helper here instead, since every text value (name, content, preferredUsername, tags) can have this form and we have helpers for similar JSON-LD conventions (`equals_or_includes`, `value_or_id`)

Reply:
This only comes up twice so I disagree about the need to abstract it.

Reply:
It's not because of DRY reasons that I'm saying it should be abstracted, it's because it's a fundamental part of processing JSON-LD content that can be implemented/tested/used independently of any actual activity processing.